### PR TITLE
CODETOOLS-7903476: JMH: Threading model should not assume stable threads

### PR DESCRIPTION
**Pull request by shipilev**
See the bug for the original bug report.

This PR makes sure the harness code is able to work with different styles of executors, including the custom executor that always gives us a new thread. Integration tests now run with several important executor modes to catch errors early.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Issue
 * [CODETOOLS-7903476](https://bugs.openjdk.org/browse/CODETOOLS-7903476): JMH: Threading model should not assume stable threads

### Reviewers
 * [Sergey Kuksenko](https://openjdk.org/census#skuksenko) (@kuksenko - Committer) ⚠️ Review applies to [4b971d34](https://git.openjdk.org/jmh/pull/107/files/4b971d347436b79b455e3f661db88c852dd83e8e)

### Contributors
 * Sergey Kuksenko `<skuksenko@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jmh.git pull/107/head:pull/107` \
`$ git checkout pull/107`

Update a local copy of the PR: \
`$ git checkout pull/107` \
`$ git pull https://git.openjdk.org/jmh.git pull/107/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 107`

View PR using the GUI difftool: \
`$ git pr show -t 107`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jmh/pull/107.diff">https://git.openjdk.org/jmh/pull/107.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jmh/pull/107#issuecomment-1572700414)

### CODE REVIEW

**Comment by shipilev** [comment]
I think this is ready for integration. What do you think, @kuksenko?

**Comment by shipilev** [comment]
Discussing which executor types we should ship with JMH is in #108. This PR would still be useful for FJP, CUSTOM, and virtual executor, AFAICS.

File: jmh-core-it/src/main/java/org/openjdk/jmh/it/CustomExecutorService.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
```
Comment (kuksenko) [inline jmh-core-it/src/main/java/org/openjdk/jmh/it/CustomExecutorService.java:2]: * Copyright Amazon.com Inc. or its affiliates.

Is that how it should be?

**Review by kuksenko — APPROVED** [review]

File: jmh-core-it/src/main/java/org/openjdk/jmh/it/CustomExecutorService.java (hunk 1 above)
Comment (shipilev) [reply jmh-core-it/src/main/java/org/openjdk/jmh/it/CustomExecutorService.java:2]: Yes, that's our standard header. Deliberately year-less.

**Pull request closed without merging**